tardis/energy_input/gamma_ray_channel.py

```diff
@@ -0,0 +1,169 @@
+import logging
+import numpy as np
+import pandas as pd
+import astropy.units as u
+import radioactivedecay as rd
+
+
+from tardis.energy_input.energy_source import (
+    get_nuclear_lines_database,
+)
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def create_isotope_dicts(raw_isotope_abundance, cell_masses):
+    """
+    Function to create a dictionary of isotopes for each shell with their masses.
+
+    Parameters
+    ----------
+    raw_isotope_abundance : pd.DataFrame
+        isotope abundance in mass fractions.
+    cell_masses : numpy.ndarray
+        shell masses in units of g
+
+    Returns
+    -------
+        isotope_dicts : Dict
+            dictionary of isotopes for each shell with their ``masses``.
+            For eg: {0: {(28, 56): {'Ni56': 0.0001}, (27, 57): {'Co56': 0.0001}}
+                    {1: {(28, 56): {'Ni56': 0.0001}, (27, 57): {'Co56': 0.0001}}} etc
+
+    """
+    isotope_dicts = {}
+    for i in range(len(raw_isotope_abundance.columns)):
+        isotope_dicts[i] = {}
+        for (
+            atomic_number,
+            mass_number,
+        ), abundances in raw_isotope_abundance.iterrows():
+            nuclear_symbol = f"{rd.utils.Z_to_elem(atomic_number)}{mass_number}"
+            isotope_dicts[i][nuclear_symbol] = (
+                abundances[i] * cell_masses[i].to(u.g).value
+            )
+
+    return isotope_dicts
+
+
+def create_inventories_dict(isotope_dict):
+    """Function to create dictionary of inventories for each shell
+
+    Parameters
+    ----------
+    isotope_dict : Dict
+        dictionary of isotopes for each shell with their ``masses``.
+
+    Returns
+    -------
+        inv : Dict
+            dictionary of inventories for each shell
+            For eg: {0: {'Ni56': <radioactivedecay.Inventory>,
```

Comment:
I don't think the inventories are separated by isotope; the inventory contains all the isotopes